tests/testsuite/clean.rs

```diff
@@ -420,21 +442,25 @@ fn clean_verbose() {
     p.cargo("build").run();
     let mut expected = String::from(
         "\
-[REMOVING] [..]target/debug/.fingerprint/bar[..]
-[REMOVING] [..]target/debug/deps/libbar[..].rlib
-[REMOVING] [..]target/debug/deps/bar-[..].d
-[REMOVING] [..]target/debug/deps/libbar[..].rmeta
+[REMOVING] [ROOT]/foo/target/debug/.fingerprint/bar-[HASH]
+[REMOVING] [ROOT]/foo/target/debug/deps/libbar-[HASH].rlib
+[REMOVING] [ROOT]/foo/target/debug/deps/bar-[HASH].d
+[REMOVING] [ROOT]/foo/target/debug/deps/libbar-[HASH].rmeta
 ",
     );
     if cfg!(target_os = "macos") {
         // Rust 1.69 has changed so that split-debuginfo=unpacked includes unpacked for rlibs.
         for obj in p.glob("target/debug/deps/bar-*.o") {
-            expected.push_str(&format!("[REMOVING] [..]{}\n", obj.unwrap().display()));
+            let obj = obj.unwrap().display().to_string();
+            expected.push_str(&format!(
+                "[REMOVING] {}\n",
+                obj.replace(p.root().to_str().unwrap(), "[ROOT]/foo")
```

Comment:
Replacing root is not enough and there is only 1 of 2 hashes in filename being redacted. How about leave it as is for now?

```
- [REMOVING] [ROOT]/foo/target/debug/deps/bar-966252632dc060f2.bar.cce207b4909ed527-cgu.0.rcgu.o
+ [REMOVING] [ROOT]/foo/target/debug/deps/bar-[HASH].bar.cce207b4909ed527-cgu.0.rcgu.o
```

Reply:
Yeah fine we that. I don't think Cargo cares about codegen unit artifacts. What about making it conditional, something like this?

```rust
#[cfg(not(target_os = "macos"))]
let expected = str![[r#"
[REMOVING] [ROOT]/foo/target/debug/.fingerprint/bar-[HASH]
[REMOVING] [ROOT]/foo/target/debug/deps/libbar-[HASH].rlib
[REMOVING] [ROOT]/foo/target/debug/deps/bar-[HASH].d
[REMOVING] [ROOT]/foo/target/debug/deps/libbar-[HASH].rmeta
[REMOVED] [FILE_NUM] files, [FILE_SIZE]B total

"#]];
#[cfg(target_os = "macos")]
let expected = str![[r#"
[REMOVING] [ROOT]/foo/target/debug/.fingerprint/bar-[HASH]
[REMOVING] [ROOT]/foo/target/debug/deps/libbar-[HASH].rlib
[REMOVING] [ROOT]/foo/target/debug/deps/bar-[HASH].d
[REMOVING] [ROOT]/foo/target/debug/deps/libbar-[HASH].rmeta
[REMOVING] [ROOT]/foo/target/debug/deps/bar-[HASH][..].o
[REMOVED] [FILE_NUM] files, [FILE_SIZE]B total

"#]];

Reply:
looks promising, but is there only one .o file?

Reply:
Nice catch. Could also just count the nubmer of `bar-.*.o` and push the same number of `[REMOVING] [ROOT]/foo/target/debug/deps/bar-[HASH][..].o` line.